examples/react/basic-ssr-streaming-file-based/src/routes/index.tsx

```diff
@@ -20,11 +18,9 @@ function IndexComponent() {
     <div className="p-2">
       <h3>Welcome Home!</h3>
       <p>Data: {data.date.getDate()}</p>
-      <React.Suspense fallback={'Loading...'}>
-        <Await promise={data.deferred}>
-          {(data) => <p>Deferred: {data.date.getDate()}</p>}
-        </Await>
-      </React.Suspense>
+      <Await promise={data.deferred} fallback={'Loading...'}>
```

Comment:
Would `Await` also work here if the promise was the sole return (both functionality and on a type-level)?

```tsx
export const Route = createFileRoute('/')({
  loader: () => new Promise<{ date: Date }>((r) => {
    setTimeout(() => r({ date: new Date() }), 1000)
  })
})

function IndexComponent() {
  // ...
      <Await promise={useLoaderDataPromise} fallback={'Loading...'}>
        {(data) => <p>Deferred: {data..getDate()}</p>}
      </Await>
  </div>
}

```

Reply:
Yes it would